File: Python/calculoAvancado.py

num1 = float(input("Digite um número para ser calculado: "))
num2 = float(input("Digite outro número para ser calculado: "))
operador = input('Digite o operador de sua preferência, "+","-","*","/","**": ')


def operar(n1,n2,operacao):
    if(operacao == "+"):
        return soma(n1,n2)
    elif(operacao == "-"):
        return subtracao(n1,n2)
    elif(operacao == "*"):
        return multiplicacao(n1,n2)
    elif(operacao == "/"):
        return divisao(n1,n2)
    elif(operacao == "**"):
        return exponenciacao(n1,n2)
    else:
        return "Erro interno operação não executada!"

def soma(x,y):
    return x + y

def subtracao(x,y):
    return x - y

def multiplicacao(x,y):
    return x * y

def divisao(x,y):
    return x / y

def exponenciacao(x,y):
    return x**y

print(f'O resultado do calculo é {operar(num1, num2, operador)}!')

File: JavaScript/calculoAvancado.js

let num1;
let num2;
let operador;

num1 = prompt(`Digite um número para ser calculado: `);
num2 = prompt(`Digite outro valor para ser calculado: `);
operador = prompt(`Digite o operador de sua preferência, "+","-","*","/","**": `);

console.log(alert(`O resultado do calculo é ${operar(num1, num2, operador)}!`))


function operar(n1,n2,operacao){
    if(operacao == "+"){
        return soma(n1,n2)
    } else if(operacao == "-"){
        return subtracao(n1,n2)
    } else if(operacao == "*"){
        return multiplicacao(n1,n2)
    } else if(operacao == "/"){
        return divisao(n1,n2)
    } else if(operacao == "**"){
        return exponeciacao(n1,n2)
    } else{
        return "Erro interno operação não executada!"
    }
}

function soma(x,y){
    return x + y
}

function subtracao(x,y){
    return x - y
}

function multiplicacao(x,y){
    return x * y
}

function divisao(x,y){
    return x / y
}

function exponeciacao(x,y){
    return x**y
}